typings/middleware-handler.d.ts

```diff
@@ -0,0 +1,12 @@
+declare module 'middleware-handler' {
+	import * as net from "net";
+
+	type Middleware = (req: Request, cltSocket: net.Socket, head: Buffer, next: () => void) => void;
+
+	class MiddlewareHandler {
+		use(middleware: Middleware): void;
+		handle(args?: any[], callback?: Function): void;
```

Comment:
Are `any[]` and `Function` the best we can do?

Reply:
Yeah, `args` can be anything and is passed verbatim to installed middleware and `callback` is called as `callback.apply(null, args)` so the signature is unknown at runtime.